core/templates/pages/contributor-dashboard-page/contributor-badge/contributor-badge.component.ts

```diff
@@ -0,0 +1,300 @@
+      (code) => {
+        const languageDescription = this.languageUtilService
+          .getAudioLanguageDescription(
+            code);
+        this.totalTranslationStats[languageDescription] = {
+          language: this.languageUtilService.getShortLanguageDescription(
+            languageDescription),
+          submissions: 0,
+          reviews: 0,
+          corrections: 0
+        };
+        this.reviewableLanguages.push(languageDescription);
+      });
+    this.userCanReviewQuestionSuggestions = (
+      userContributionRights.can_review_questions);
+    this.userCanSuggestQuestions = (
+      userContributionRights.can_suggest_questions);
+    this.userHasQuestionRights = (
+      this.userCanSuggestQuestions ||
+      this.userCanReviewQuestionSuggestions);
+
+    const response = await this.contributionAndReviewStatsService.fetchAllStats(
+      username);
+
+    if (response.translation_contribution_stats.length > 0) {
+      await response.translation_contribution_stats.map((stat) => {
+        const languageDescription =
+          this.languageUtilService.getAudioLanguageDescription(
+            stat.language_code);
+
+        // There can be languages that the contributor has translation
+        // contribution stats but not translation review stats. Hence we need to
+        // initialize TotalTranslationStats objects for those languages.
+        if (!this.totalTranslationStats[languageDescription]) {
+          this.totalTranslationStats[languageDescription] = {
+            language: this.languageUtilService.getShortLanguageDescription(
+              languageDescription),
+            submissions: stat.accepted_translations_count,
+            reviews: 0,
+            corrections: 0
+          };
+        } else {
+          this.totalTranslationStats[languageDescription].submissions += (
+            stat.accepted_translations_count);
+        }
+      });
+    }
+    if (response.translation_review_stats.length > 0) {
+      await response.translation_review_stats.map((stat) => {
+        const languageDescription =
+          this.languageUtilService.getAudioLanguageDescription(
+            stat.language_code);
+
+        this.totalTranslationStats[languageDescription].reviews += (
+          stat.reviewed_translations_count);
+        this.totalTranslationStats[languageDescription].corrections += (
+          stat.accepted_translations_with_reviewer_edits_count);
+      });
+    }
+    if (response.question_contribution_stats.length > 0) {
+      response.question_contribution_stats.map((stat) => {
+        this.totalQuestionStats.submissions = stat.accepted_questions_count;
+      });
+    }
+    if (response.question_review_stats.length > 0) {
+      response.question_review_stats.map((stat) => {
+        this.totalQuestionStats.reviews = stat.reviewed_questions_count;
+        this.totalQuestionStats.corrections = (
+          stat.accepted_questions_with_reviewer_edits_count);
+      });
+    }
```

Comment:
Don't you also need to add these counts since there is one question stat per topic? Also, there should be unit tests verifying these counts/obtained badges.